tap_snowflake/sync_strategies/common.py

```diff
@@ -111,6 +122,16 @@ def generate_select_sql(catalog_entry, columns):
             escaped_columns.append(f'TO_VARCHAR({escaped_col}) as {escaped_col}')
         elif property_format == 'geography':
             escaped_columns.append(f'ST_ASTEXT({escaped_col}) as {escaped_col}')
+        # Castings below were added for WP-21311 to make sure that Snowflake import using s3 unload sync vs regular sync 
+        # uses the same select query.
+        elif property_format == 'time':
+            escaped_columns.append(f"TO_VARCHAR({escaped_col}, 'HH24:MI:SS.FF6') as {escaped_col}")
```

Comment:
Using regular sync, time columns are imported as text with precision up to microsecond. However, inn case the data only has value up to second, it was imported without trailing 0. I think this is a bug on Snowflake side because default time output format was HH24:MI:SS, which should only give us up to seconds.
![image](https://github.com/symon-ai/pipelinewise-tap-snowflake/assets/86244437/44be6658-5495-4b24-a9c3-007e16d50aa0)

Using Copy method, Snowflake was returning values with precision up to second. For consistency, casting to string with format up to microsecond. This will lead to have trailing zeros at the end. 
<img width="207" alt="image" src="https://github.com/symon-ai/pipelinewise-tap-snowflake/assets/86244437/5de3d485-6bec-4909-a652-6b2f63bed02d">

Other option: All timestamp columns store precision up to millisecond, so we could use the same for time.